packages/mongodb-log-writer/src/mongo-log-manager.ts

```diff
@@ -101,18 +123,25 @@ export class MongoLogManager {
         // a large number of machines using a shared network-mounted $HOME directory
         // where lots and lots of log files end up and filesystem operations happen
         // with network latency.
-        if (Date.now() - deletionStartTimestamp > maxDurationMs) break;
+        if (Date.now() - deletionStartTimestamp > maxDurationMs) {
+          break;
+        }
+
+        if (!dirent.isFile()) {
+          continue;
+        }
```

Comment:
This leaves us in a situation where, if removing a directory fails for any reason, we're never going to clean that directory up. I think that may be something we just need to accept, unless we want to start e.g. imposing naming restrictions on the nested subdirectories – that's something we should probably decide on now, though.